library/core/src/std/casting.fe

```diff
@@ -0,0 +1,43 @@
+extern {
+    pub fn todo() -> !
+}
+
+pub trait IntUpcast<To> {
+    fn upcast(self) -> To
+}
+
+pub impl IntUpcast<u256> for u8 {
+    fn upcast(self) -> u256 {
+        todo()
+    }
+}
+
+pub impl IntUpcast<u256> for usize {
+    fn upcast(self) -> u256 {
+        todo()
+    }
+}
+
+pub trait IntDowncast<To> {
+    fn downcast_truncate(self) -> To
+    fn downcast_saturate(self) -> To
+    // fn downcast(self) -> Option<To> // Option is in core lib, which isn't visible in lsp (see grant's PR)
+}
+
+pub impl IntDowncast<u8> for u256 {
```

Comment:
Question 2:
In this PR I didn't include any EVM, nor Solidy specific code. Should that be maintained in a separated repo? If so, should I also remove all u256 references from the core library?